src/phoenix/experimental/evals/long_context_evaluators.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Optional
+
+from phoenix.experimental.evals import PromptTemplate
+from phoenix.experimental.evals.models import BaseEvalModel
+
+
+class Refiner:
+    def __init__(
+        self,
+        model: BaseEvalModel,
+        initial_prompt_template: PromptTemplate,
+        refine_prompt_template: PromptTemplate,
+        reduce_prompt_template: Optional[PromptTemplate] = None,
```

Comment:
I just caught that this is "optional" - but in general I don't think I understand the need for a final prompt in the refine strategy as it's already aggregating as it goes. Can you help me understand why this final prompting would be necessary?

Also question about the param - I think since it's really not reducing (e.g. it's just taking the accumulator) it's not actually reducing anything so I'd consider naming it something like `synthesis` if necessary.

Reply:
That last step is included here in case the user wants to do classification and get a single class at the end.